src/IonParserTextRaw.ts

```diff
@@ -1510,18 +1527,23 @@ private _test_symbol_as_annotation() : boolean {
     return ch;
   }
 
-  private _read_N_digits(n: number) : number {
-    var ch, ii=0;
-    while ( ii < n ) {
-      ch = this._read();
-      if (!IonText.is_digit(ch)) {
-        this._error( ""+n+" digits required "+ii+" found" );
-        return ERROR;
-      }
-      ii++;
+    private _readNDigits(n: number) : number {//This is clearly bugged it reads n + 1 digits.
+        let ch : number
+        if (n <= 0) throw new Error("Cannot read a negative number of digits.");
+        while(n--) {
+            if (!IonText.is_digit(ch = this._read())) throw new Error("Expected digit, got: " + ch);
+        }
+        return ch;
+    }
+
+    private _readPastNDigits(n: number) : number {//This is clearly bugged it reads n + 1 digits.
+        let ch : number
+        if (n <= 0) throw new Error("Cannot read a negative number of digits.");
+        while(n--) {
+            if (!IonText.is_digit(ch = this._read())) throw new Error("Expected digit, got: " + ch);
+        }
+        return this._read();
```

Comment:
Is this extra `this.read()` that causes the n+1 error? Is this the intention of the method `_readPastNDigits` and also the reason it is different from `_readNDigits`? 

Can we add documentation to both so that we know what the intention is? Also seems to me that you could refactor the code a bit, the difference seems to be on what they return but the main loop is shared.

Reply:
Isn't _readPastNDigits just:

```js
private _readPastNDigits(n: number) : number {
    _readNDigits(n);
    return this._read();
}
```

Reply:
The difference being one accepts none digits as the terminating character.
I agree, ReadPastNDigits should be refactored to 


private _readPastNDigits(n: number) : number {
    _readNDigits(n);
    return this._read();
}

Reply:
done